persist/fs/index_write.go

```diff
@@ -97,46 +97,35 @@ func (w *indexWriter) Open(opts IndexWriterOpenOptions) error {
 	var (
 		namespace  = opts.Identifier.Namespace
 		blockStart = opts.Identifier.BlockStart
-		err        error
 	)
 	w.err = nil
 	w.blockSize = opts.BlockSize
 	w.start = blockStart
 	w.fileSetType = opts.FileSetType
+	w.volumeIndex = opts.Identifier.VolumeIndex
 	w.snapshotTime = opts.Snapshot.SnapshotTime
 	w.segments = nil
 
 	switch opts.FileSetType {
 	case persist.FileSetSnapshotType:
 		w.namespaceDir = NamespaceIndexSnapshotDirPath(w.filePathPrefix, namespace)
 		// Can't do this outside of the switch statement because we need to make sure
-		// the directory exists before calling NextSnapshotFileIndex
+		// the directory exists before calling NextSnapshotFileSetIndex
```

Comment:
minor nit: I think you can simplify the code a little now:
```
switch opts.FileSetType{
case persist.FileSetSnapshotType:
 w.namespaceDir = ...
case FileSetFlushType:
  w.namespaceDir = ...
}
if err := os.MkdirAll(...) ; err !=nil {
...
}
```

Reply:
Makes sense, cheers.